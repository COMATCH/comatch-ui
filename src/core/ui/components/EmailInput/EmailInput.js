import React from 'react';
import { TextInput } from '../TextInput';

/* eslint-disable react/prop-types */
export const EmailInput = ({ type, ...props }) => <TextInput type="email" {...props} />;
