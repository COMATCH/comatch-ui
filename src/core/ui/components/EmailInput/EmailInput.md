
## EmailInput

From [`EmailInput.js`](EmailInput.js)



prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------



