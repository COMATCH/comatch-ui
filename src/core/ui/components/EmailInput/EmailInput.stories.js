import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { EmailInput } from './EmailInput';

storiesOf('EmailInput', module)
    .add('with basic style', () => <EmailInput name="field" onChange={action('changed')} />)
    .add('with placeholder', () => (
        <EmailInput name="field" onChange={action('changed')} placeholder="Placeholder text" />
    ))
    .add('with value', () => <EmailInput name="field" onChange={action('changed')} value="Value text" />)
    .add('with label', () => <EmailInput name="field" onChange={action('changed')} label="Label text" />)
    .add('with error', () => (
        <EmailInput name="field" onChange={action('changed')} label="Label text" inputError="Error text" />
    ));
