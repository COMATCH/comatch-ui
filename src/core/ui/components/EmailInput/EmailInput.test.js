import React from 'react';
import renderer from 'react-test-renderer';
import { EmailInput } from './EmailInput';
import '../../../config/tests/setup';

describe('EmailInput', () => {
    it('should render correctly', () => {
        const mockFunction = jest.fn();

        const emailInput = renderer.create(
            <EmailInput name="email-name-sample" onChange={mockFunction}/>
        ).toJSON();

        expect(emailInput).toMatchSnapshot();
    });
});
