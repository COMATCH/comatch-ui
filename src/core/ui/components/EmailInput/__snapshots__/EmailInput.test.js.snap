// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput should render correctly 1`] = `
<div
  className="Input TextInput block"
  id={null}
>
  
  <input
    autoComplete={null}
    className="TextInput__input"
    disabled={null}
    max={null}
    maxLength={null}
    min={null}
    name="email-name-sample"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onClick={[Function]}
    onFocus={[Function]}
    placeholder=""
    type="email"
    value=""
  />
  
</div>
`;
